var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -394,6 +394,11 @@ def cmake_args(self):
             # default is off.
             options.append("-DLegion_USE_Kokkos=ON")
             os.environ["KOKKOS_CXX_COMPILER"] = spec["kokkos"].kokkos_cxx
+            if spec.satisfies("+cuda+cuda_unsupported_compiler ^kokkos%clang +cuda"):
+                # Keep CMake CUDA compiler detection happy
+                options.append(
+                    self.define("CMAKE_CUDA_FLAGS", "--allow-unsupported-compiler -std=c++17")
```

Comment:
I just want to double check. Is this two flags or one? I was pretty sure we were inferring the CUDA standard from the CXX one:

https://github.com/StanfordLegion/legion/blob/3f27977943626ef23038ef0049b7ad1b389caad1/CMakeLists.txt#L477

calls:

https://github.com/StanfordLegion/legion/blob/3f27977943626ef23038ef0049b7ad1b389caad1/cmake/cuda_helpers.cmake#L58-L66

and used here:

https://github.com/StanfordLegion/legion/blob/3f27977943626ef23038ef0049b7ad1b389caad1/runtime/CMakeLists.txt#L289

Presumably we'd want to use the existing standard passing mechanism unless that was broken somehow.

Reply:
For context, this is the kind of error you get without both:
```
1 error found in build log:                                                                                                                                                                  
     29    Call Stack (most recent call first):                                                                                                                                              
     30      .../spack/opt/spack/linux-sles15-zen2/clang-17.0.6/kokkos-4.3.01-v4fi7xdtklfieu7akpjzlr4gvu3cc2w2/lib64/cmake/Kokkos/KokkosConfig.cmake:56 (INCLUDE)                                                                                                                                                                  
     31      CMakeLists.txt:327 (find_package)                                                                                                                                               
     32                                                                                                                                                                                      
     33                                                                                                                                                                                      
     34    -- Kokkos version: 4.3.1                                                                                                                                                          
  >> 35    CMake Error at .../spack/opt/spack/linux-sles15-zen2/clang-17.0.6/cmake-3.27.9-vaq7ebhsdssd7teu62gqmb2hhzpn6x2i/share/cmake-3.27/Modules/CMakeDetermineCompilerId.cmake:753 (message):                                                                                                                                  
     36      Compiling the CUDA compiler identification source file                                                                                                                          
     37      "CMakeCUDACompilerId.cu" failed.                                                                                                                                                
     38                                                                                                                                                                                      
     39      Compiler:                                                                                                                                                                       
     40      .../spack/opt/spack/linux-sles15-zen2/clang-17.0.6/cuda-11.7.1-da5pf3jkni3wor2meccydgf6u377cpdb/bin/nvcc 
```
So this happens prior to all that logic, during `find_package(Kokkos)`.

Reply:
So if you wanted to fix it in your own CMake, you'd probably be forced to first do all the `CMAKE_CUDA_FLAGS`, `CMAKE_CUDA_STANDARD` machinery prior to calling `find_package(Kokkos)`. It has to happen prior to anyone enabling CUDA as a language or testing for it.

Reply:
I've also tried only using `CMAKE_CUDA_FLAGS=--allow-unsupported-compiler` on its own, tried to additionally set `CMAKE_CUDA_STANDARD=17`, but only `CMAKE_CUDA_FLAGS` seems to come through.

Reply:
@elliottslaughter here is the error log without `-std=c++17`
[error.log](https://github.com/user-attachments/files/16090719/error.log)

Reply:
Ok. Let's take your patch here for now, but please also submit an upstream Legion issue so we can investigate what a proper fix would be.

Reply:
https://github.com/StanfordLegion/legion/issues/1709